### PR TITLE
[DOCS] Service account edits

### PR DESCRIPTION
**Pull request by lockewritesdocs**
General edits for the service account docs, in addition to:

* Putting all of the content on a single page to increase readability and enable users to search for content they want.
* De-emphasizing the connection to built-in accounts. [This comment](https://github.com/elastic/elasticsearch/pull/71729#discussion_r630794724) from Tim was was lingering in my brain, and I do think it makes sense to decouple service accounts as an evolution of the built-in users. It's more of an implementation detail IMO.

Preview link: https://elasticsearch_73732.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/service-accounts.html

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

File: x-pack/docs/en/security/authentication/service-accounts.asciidoc [hunk 1]
```diff
@@ -34,13 +37,20 @@ Service accounts are not included in the response of the
...
 
 `elastic/fleet-server`:: The service account used by the {fleet} server to
 communicate with {es}.
 
+While not recommended, it is possible to create a regular user with the same
+username as a service account principal. There will be no security complications
+because the {es} views them as distinct, separate identities. In addition, a service
+account can only be authenticated with service tokens which is not applicable for
+regular users.
+
```
Comment (lockewritesdocs) [inline x-pack/docs/en/security/authentication/service-accounts.asciidoc:53]: @ywangd, this information contradicts the note that follows. If we don't recommend that users create a user with the same username as a service account principal, I think we should actively discourage this behavior. I think we can remove this text and amend the following note to read:

> IMPORTANT: Do not attempt to use service accounts for authenticating individual users. Service accounts can only be authenticated with service tokens, which are not applicable to regular users.

Comment (ywangd) [reply x-pack/docs/en/security/authentication/service-accounts.asciidoc:53]: I agree that we should discourage it. I was just trying to be comprehensive. But it does add to confusion as a side effect. I am happy for you to amend it as you see fit.

**Comment by lockewritesdocs** [comment]
@ywangd, thanks for your recent edits. I think that this information is ready for @tvernum to review.

**Review by tvernum — APPROVED** [review]
LGTM

**Review by ywangd — APPROVED** [review]

**Pull request merged**